centreon/src/Core/Notification/Infrastructure/Repository/DbHostGroupResourceRepository.php

```diff
@@ -0,0 +1,413 @@
+
+        $statement = $this->db->prepare($this->translateDbName($concatenator->concatAll()));
+        $concatenator->bindValuesToStatement($statement);
+        $statement->execute();
+
+        $resources = array_map(
+            (fn($data) => new NotificationGenericObject($data['hg_id'], $data['hg_name'])),
+            $statement->fetchAll(\PDO::FETCH_ASSOC)
+        );
+
+        return new NotificationResource(
+            self::RESOURCE_TYPE,
+            self::EVENT_ENUM,
+            $resources,
+            NotificationHostEventConverter::fromBitFlags($hostgroupEvents),
+            NotificationServiceEventConverter::fromBitFlags($includedServiceEvents),
+        );
+    }
+
+    /**
+     * @inheritDoc
+     */
+    public function add(int $notificationId, NotificationResource $resource): void
+    {
+        $this->info(
+            'Add resource',
+            [
+                'resource_type' => self::RESOURCE_TYPE,
+                'notification_id' => $notificationId,
+                'resource' => $resource,
+            ]
+        );
+
+        $alreadyInTransaction = $this->db->inTransaction();
+        if (! $alreadyInTransaction) {
+            $this->db->beginTransaction();
+        }
+
+        $statement = $this->db->prepare($this->translateDbName(
+            <<<'SQL'
+                UPDATE `:db`.notification SET
+                    hostgroup_events = :events,
+                    included_service_events = :serviceEvents
+                WHERE id = :notificationId
+                SQL
+        ));
+        $statement->bindValue(
+            ':events',
+            (self::EVENT_ENUM_CONVERTER)::toBitFlags($resource->getEvents()),
+            \PDO::PARAM_INT
+        );
+        $statement->bindValue(':notificationId', $notificationId, \PDO::PARAM_INT);
+        $statement->bindValue(
+            ':serviceEvents',
+            method_exists($resource, 'getServiceEvents')
+                ? NotificationServiceEventConverter::toBitFlags($resource->getServiceEvents())
+                : 0,
+            \PDO::PARAM_INT
+        );
+        $statement->execute();
+
+        $subQuery = [];
+        $bindElem = [];
+        foreach ($resource->getResources() as $key => $resourceElem) {
+            $subQuery[] = "(:notificationId, :resource_{$key})";
+            $bindElem[":resource_{$key}"] = $resourceElem->getId();
+        }
+        $statement = $this->db->prepare($this->translateDbName(
+            'INSERT INTO `:db`.notification_hg_relation (notification_id, hg_id) VALUES ' . implode(', ', $subQuery)
+        ));
+        $statement->bindValue(':notificationId', $notificationId, \PDO::PARAM_INT);
+        foreach ($bindElem as $key => $value) {
+            $statement->bindValue($key, $value, \PDO::PARAM_INT);
+        }
+        $statement->execute();
+
+        if (! $alreadyInTransaction) {
+            $this->db->commit();
+        }
```

Comment:
No rollback ?
+
This is already called in a transaction through the usecase with an injected `DataStorageEngineInterface`